src/main/java/org/opensearch/knn/plugin/script/KNNScoringScriptEngine.java

```diff
@@ -39,7 +39,7 @@ public <FactoryType> FactoryType compile(String name, String code, ScriptContext
             KNNCounter.SCRIPT_COMPILATION_ERRORS.increment();
             throw new IllegalArgumentException("Unknown script name " + code);
         }
-        ScoreScript.Factory factory = KNNScoreScriptFactory::new;
```

Comment:
do we need this change?

Reply:
Yes we need, since KNNScoreScriptFactory does not have a no-argument constructor that matches what's expected by the interface, we can't use the method reference `KNNScoreScriptFactory::new` here. And the IDE 
will report error "Cannot resolve constructor 'KNNScoreScriptFactory'"